src/poc/miner_poc_statem.erl

```diff
@@ -74,7 +74,8 @@
     mining_timeout = ?MINING_TIMEOUT :: non_neg_integer(),
     retry = ?CHALLENGE_RETRY :: non_neg_integer(),
     receipts_timeout = ?RECEIPTS_TIMEOUT :: non_neg_integer(),
-    poc_restarts = ?POC_RESTARTS :: non_neg_integer()
+    poc_restarts = ?POC_RESTARTS :: non_neg_integer(),
+    targeting_data :: {atom(), blockchain_block:hash(), binary()} | undefined
```

Comment:
Isn't the third parameter here a rocksdb snapshot? This isn't safe to serialize to disk as it will not point to anything after a node restart...?

Reply:
I think you need to store the block height of the ledger here, so you can use ledger_at to obtain a ledger snapshot when deserializing the stored state

Reply:
Well spotted! now fixed...